### ISSUE TITLE
Feature request: Additional informations on entities

### ISSUE THREAD

**Issue by gtteamamxx** [alliedmodders/amxmodx#570]
I was wondering, what about storing custom data for entitiy instead creating Trie?

If we want to keep data simple for some entity, already we are using

`EV_INT_iuser1, EV_INT_iuser2` etc.  because creating variable

`new entities[4096]` is bad idea

What about storing DataPacks for entity and auto-releasing it when entity is removing by remove_entity?

some natives like:
`native entity_set_data(entityId, DataPack: data);` and` entity_get_data`

**Comment by WPMGPRoSToTeMa** [comment]
[Module: Custom Entity Data](https://forums.alliedmods.net/showthread.php?t=294188)

**Comment by gtteamamxx** [comment]
@WPMGPRoSToTeMa 
Yeah, i know. I was thinking about adding it to amxmodx like default feature.
maybe @rsKliPPy can marge this 💯

**Comment by Arkshine** [comment]
That would be actually a nice addition. There are a lot of plugins trying to set data per entity using fields which are unused or irrelevant but can conflict easily with others plugins or others mods.

**Comment by gtteamamxx** [comment]
@Arkshine 
Believe me, i'm using whatever i can to store data without creating global tries/variables ;P
![obraz](https://user-images.githubusercontent.com/10392319/45384302-e9508b00-b60d-11e8-8ac4-5e7187e68d7c.png)

**Comment by IgnacioFDM** [comment]
I think numerical keys that you have to initialize are "better" than string keys, but depends on the situation of course.

**Comment by gtteamamxx** [comment]
@IgnacioFDM 
Yes, it depends on the situation.
I made some time ago a plugin for assisting of killing a monster.
I mean - more than one player can shot at the monster, if player total damage was more than X percentage, then i'm giving some benefits to him.
So, i have to
1) Store on monster entity id all player id's to know which player shot to this monster
2) Store all monster entity id on player's array (inverse of 1) )

I think that better is option number 1, so, i did this:

![image](https://user-images.githubusercontent.com/10392319/45485397-c92ce300-b757-11e8-8730-6045a83506c0.png)


I have a Trie with a monster entity id's as Trie index
To each Trie Cell of a monster i'm creating an Array to store information about 
a) Player Id
b) Player damage

When monster is killed i'm retrieving a Trie from this monster and iterating on all Arrays assigned to this Trie.

It's working like a charm, but managing this code is so hard, but this is what programming is like. 
I have to be aware memory leaks. Creating something like
entity_set_data will bo so powerful.

out of topic: If you want to see full code: https://github.com/gtteamamxx/Tower-Defense/blob/master/addons/amxmodx/scripting/td_assist.sma

**Comment by IgnacioFDM** [comment]
If you wanted, you could implement something identical to @rsKliPPy module in pawn. It's basically just an array of tries, of size ```maxEntities```

**Comment by gtteamamxx** [comment]
I think imho that better will be implementing something like entity_set_data.

**Comment by IgnacioFDM** [comment]
Yes of course. I'm just saying if you wanted something that works right now, and compatible with AMXX <= 1.9
